src/components/AddPeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -0,0 +1,280 @@
+    this.currentOrgPerson.type = type
+    this.activeIndex = -1
+    this.nextId = (this.orgPersonList.length === 0) ? 0 : this.orgPersonList[this.orgPersonList.length - 1].id + 1
+    this.addEditInProgress = true
+    this.showOrgPersonForm = true
+  }
+
+  private editOrgPerson (index: number) : void {
+    this.currentOrgPerson = { ...this.orgPersonList[index] }
+    this.activeIndex = index
+    this.addEditInProgress = true
+    this.showOrgPersonForm = true
+  }
+
+  private onAddEditOrgPerson (person: OrgPersonIF): void {
+    let newList: OrgPersonIF[] = Object.assign([], this.orgPersonList)
+    // New Person.
+    if (this.activeIndex === -1) {
+      newList.push(person)
+    } else {
+    // Edit Person.
+      newList.splice(this.activeIndex, 1, person)
+    }
+    this.setOrgPersonList(newList)
+    this.setAddPeopleAndRoleStepValidity(this.hasValidRoles())
+    this.resetData()
+  }
+
+  private onRemovePerson (index: number) : void {
+    let newList: OrgPersonIF[] = Object.assign([], this.orgPersonList)
+    newList.splice(index, 1)
+    this.setOrgPersonList(newList)
+    this.setAddPeopleAndRoleStepValidity(this.hasValidRoles())
+    this.resetData()
+  }
+
+  private removeCompletingPartyAssignment () {
+    let newList: OrgPersonIF[] = Object.assign([], this.orgPersonList)
+    const cpList = newList.filter(people => people.roles.includes(Roles.COMPLETING_PARTY))
+    if (cpList.length > 0) {
+      let completingParty : OrgPersonIF = cpList[0]
+      completingParty.roles = completingParty.roles.filter(role => role !== Roles.COMPLETING_PARTY)
+      this.setOrgPersonList(newList)
+    }
+  }
+
+  private resetData (): void {
+    this.currentOrgPerson = null
+    this.activeIndex = -1
+    this.addEditInProgress = false
+    this.showOrgPersonForm = false
+  }
+
+  private hasValidRoles () : boolean {
+    const numOfDirector: number =
+    this.orgPersonList.filter(people => people.roles.includes(Roles.DIRECTOR)).length
+    const numOfIncorporator: number =
+    this.orgPersonList.filter(people => people.roles.includes(Roles.INCORPORATOR)).length
+    const numOfCompletingParty: number =
+    this.orgPersonList.filter(people => people.roles.includes(Roles.COMPLETING_PARTY)).length
+    const numOfPeopleWithNoRoles:number =
+    this.orgPersonList.filter(people => people.roles.length === 0).length
+
+    if (this.entityFilter(EntityTypes.BCOMP)) {
+      return numOfCompletingParty === 1 && numOfIncorporator >= 1 && numOfDirector >= 1 && numOfPeopleWithNoRoles === 0
+    } else if (this.entityFilter(EntityTypes.COOP)) {
+      return numOfCompletingParty === 1 && numOfIncorporator >= 3 && numOfDirector >= 3 && numOfPeopleWithNoRoles === 0
+    }
+  }
+
+  private hasRole (roleName: string, count:number, mode:string) : boolean {
+    const orgPersonWithSpecifiedRole: OrgPersonIF[] =
+    this.orgPersonList.filter(people => people.roles.includes(roleName))
+    if (mode === 'EXACT') {
```

Comment:
If possible, declare constants or enums for mode, so that caller and callee are always in sync.

Reply:
Done